src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1246,23 +1247,10 @@ def _to_external_value(self, value: Union[int, str, float], column_type: DataTyp
     def _to_snowpark_type(self, schema: Tuple[tuple]) -> StructType:
         fields = []
         for column in schema:
-            if column[1] == int:
-                field = StructField(column[0], IntegerType(), column[6])
-            elif column[1] == float:
-                field = StructField(column[0], FloatType(), column[6])
-            elif column[1] == decimal.Decimal:
-                field = StructField(
-                    column[0], DecimalType(column[4], column[5]), column[6]
-                )
-            elif column[1] == str:
-                field = StructField(column[0], StringType(), column[6])
-            elif column[1] == datetime.datetime:
-                field = StructField(column[0], TimestampType(), column[6])
-            elif column[1] == bool:
-                field = StructField(column[0], BooleanType(), column[6])
-            else:
-                raise ValueError("unsupported type")
-
+            datatype = sql_server_type_to_snow_type(column)
+            field = StructField(
+                column[0], datatype, True if column[5].lower == "yes" else False
```

Comment:
should be lower() call?
```suggestion
                column[0], datatype, True if column[5].lower() == "yes" else False
```